firedrake/interpolation.py

```diff
@@ -304,6 +368,102 @@ def _interpolator(V, tensor, expr, subset, arguments, access):
         return copyin + (parloop, ) + copyout
 
 
+@singledispatch
+def rebuild(element, expr):
+    raise NotImplementedError(f"Cross mesh interpolation not implemented for a {element} element.")
+
+
+# Only tested with P0DG this but probably will work more generally with any
+# element that advertises a dual set fiat equivalent.
+@rebuild.register(finat.DiscontinuousLagrange)
+def rebuild_dg(element, expr):
+    # To tabulate on the given element (which is on a different mesh to the
+    # expression) we must do so at runtime. We therefore create a quadrature
+    # element with runtime points to evaluate for each point in the element's
+    # dual basis. This exists on the same reference cell as the input element
+    # and we can interpolate onto it before mapping the result back onto the
+    # target space.
+    expr_tdim = expr.ufl_domain().topological_dimension()
+    # Need 1 point evaluation and equivalent weight for each in dual basis -
+    # this uses FIAT's dual basis for now but, when implemented, should use
+    # FInAT's dual evaluation infrastructure.
+    num_points = sum(len(dual.get_point_dict()) for dual in element.fiat_equivalent.dual_basis())
+    weights = []
+    for dual in element.fiat_equivalent.dual_basis():
+        pts = dual.get_point_dict().keys()
+        for p in pts:
+            for w, _ in dual.get_point_dict()[p]:
+                weights.append(w)
+    assert len(weights) == num_points
```

Comment:
This being here makes me wonder about 
 - (a) whether this should really live here and
 - (b) how this will work, in practice, with elements with complicated dual bases

Reply:
(I'm trying to think ahead to getting this working on higher dimension meshes)

Reply:
Okay so conclusion: This is useful for thinking about future design - in the future this can be absorbed into FInAT as a transformer that lets you do run time tabulation of that element. I can comment as much here. But for now there's really no need to be using FIAT's hard-work dual basis API so just hard code what the values are in this case.